src/Nethermind/Nethermind.Core/Bloom.cs

```diff
@@ -157,35 +157,48 @@ public void Add(LogEntry[] logEntries, Bloom blockBloom = null)
             }
         }
         
-        public bool IsMatch(LogEntry logEntry)
+        public bool Matches(LogEntry logEntry)
         {
-            if (!IsMatch(logEntry.LoggersAddress.Bytes))
+            if (Matches(logEntry.LoggersAddress))
             {
-                return false;
-            }
-            
-            for (int topicIndex = 0; topicIndex < logEntry.Topics.Length; topicIndex++)
-            {
-                Keccak topic = logEntry.Topics[topicIndex];
-                if (!IsMatch(topic.Bytes))
+                for (int topicIndex = 0; topicIndex < logEntry.Topics.Length; topicIndex++)
                 {
-                    return false;
+                    if (!Matches(logEntry.Topics[topicIndex]))
+                    {
+                        return false;
+                    }
                 }
-            }
 
-            return true;
+                return true;
+            }
 
+            return false;
         }
+
+        public bool Matches(Address address) => Matches(address.Bytes);
+        
+        public bool Matches(Keccak topic) => Matches(topic.Bytes);
+        
+        public bool Matches(ref (int Index1, int Index2, int Index3) indexes) => _bits[indexes.Index1] && _bits[indexes.Index2] && _bits[indexes.Index3];
         
-        private (int Index1, int Index2, int Index3) GetIndexes(byte[] sequence)
+        public bool Matches((int Index1, int Index2, int Index3) indexes) => _bits[indexes.Index1] && _bits[indexes.Index2] && _bits[indexes.Index3];
+        
+        public bool Matches((int Index1, int Index2, int Index3)? indexes) => indexes.HasValue && _bits[indexes.Value.Index1] && _bits[indexes.Value.Index2] && _bits[indexes.Value.Index3];
+
+        public static (int Index1, int Index2, int Index3) GetIndexes(Address address) => GetIndexes(address.Bytes);
```

Comment:
yeah, I think we use it too many times not to build a struct of it

Reply:
Yeah it started small, and then grew out of hand. It leaked outside and naming it is way overdue :)